src/main/java/dao/impl/ProductsDaoImpl.java

```diff
@@ -0,0 +1,54 @@
+package dao.impl;
+
+import dao.ProductsDao;
+import database.Database;
+import model.Product;
+import org.apache.log4j.Logger;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+public class ProductsDaoImpl implements ProductsDao {
+
+    private static final Logger logger = Logger.getLogger(ProductsDaoImpl.class);
+
+    @Override
+    public void addProduct(Product product) {
+        Database.products.add(product);
+    }
+
+    @Override
+    public void addProduct(Long id, String name, String description, Double price) {
+        Product product = new Product(id, name, description, price);
+        Database.products.add(product);
+        logger.info("Product " + product + "added in DB");
+    }
+
+    @Override
+    public List<Product> getAllProducts() {
+        return Database.products;
+    }
+
+    @Override
+    public void deleteProduct(Long id) {
+        Optional<Product> optProduct = Database.products.stream()
```

Comment:
maybe better to use method getById() here?

Reply:
yes, sure